### PR TITLE
Support user defined CMAKE_INSTALL_INCLUDEDIR values

### PR DESCRIPTION
**Pull request by robertmaynard**
Needed to fix https://github.com/NVIDIA/thrust/issues/1541

### CODE REVIEW

File: cmake/CubInstallRules.cmake [hunk 1]
```diff
@@ -11,11 +11,17 @@ include(GNUInstallDirs)
 set(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY TRUE)
 
 install(DIRECTORY "${CUB_SOURCE_DIR}/cub"
-  TYPE INCLUDE
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
```
Comment (allisonvacanti) [inline cmake/CubInstallRules.cmake:14]: Add quotes around the path.

File: cub/cmake/cub-header-search.cmake [hunk 2]
```diff
@@ -0,0 +1,7 @@
+unset(_CUB_VERSION_INCLUDE_DIR CACHE) # Clear old result to force search
+find_path(_CUB_VERSION_INCLUDE_DIR cub/version.cuh
+  NO_DEFAULT_PATH # Only search explicit paths below:
+  PATHS
+    ${CMAKE_CURRENT_LIST_DIR}/../..            # Source tree
+)
+set_property(CACHE _CUB_VERSION_INCLUDE_DIR PROPERTY TYPE INTERNAL)
```
Comment (allisonvacanti) [inline cub/cmake/cub-header-search.cmake:7]: Missing newline at EOF

File: cmake/CubInstallRules.cmake [hunk 3]
```diff
@@ -11,11 +11,17 @@ include(GNUInstallDirs)
...
   FILES_MATCHING
     PATTERN "*.cuh"
 )
 
 install(DIRECTORY "${CUB_SOURCE_DIR}/cub/cmake/"
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/cub"
+  PATTERN cub-header-search EXCLUDE
 )
+configure_file("${CUB_SOURCE_DIR}/cub/cmake/cub-header-search.cmake.in"
```
Comment (allisonvacanti) [inline cmake/CubInstallRules.cmake:23]: Add a comment explaining why this is needed (similar to the Thrust PR)

File: cub/cmake/cub-config-version.cmake [hunk 4]
```diff
@@ -1,12 +1,6 @@
 # Parse version information from version.cuh:
-unset(_CUB_VERSION_INCLUDE_DIR CACHE) # Clear old result to force search
-find_path(_CUB_VERSION_INCLUDE_DIR cub/version.cuh
-  NO_DEFAULT_PATH # Only search explicit paths below:
-  PATHS
-    ${CMAKE_CURRENT_LIST_DIR}/../..            # Source tree
-    ${CMAKE_CURRENT_LIST_DIR}/../../../include # Install tree
-)
-set_property(CACHE _CUB_VERSION_INCLUDE_DIR PROPERTY TYPE INTERNAL)
+include(${CMAKE_CURRENT_LIST_DIR}/cub-header-search.cmake)
```
Comment (allisonvacanti) [inline cub/cmake/cub-config-version.cmake:2]: Add quotes around path

File: cub/cmake/cub-header-search.cmake [hunk 5]
```diff
@@ -0,0 +1,7 @@
+unset(_CUB_VERSION_INCLUDE_DIR CACHE) # Clear old result to force search
+find_path(_CUB_VERSION_INCLUDE_DIR cub/version.cuh
+  NO_DEFAULT_PATH # Only search explicit paths below:
+  PATHS
+    ${CMAKE_CURRENT_LIST_DIR}/../..            # Source tree
```
Comment (allisonvacanti) [inline cub/cmake/cub-header-search.cmake:5]: Quotes

File: cub/cmake/cub-header-search.cmake.in [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Parse version information from version.h:
+unset(_CUB_VERSION_INCLUDE_DIR CACHE) # Clear old result to force search
+find_path(_CUB_VERSION_INCLUDE_DIR cub/version.cuh
+  NO_DEFAULT_PATH # Only search explicit paths below:
+  PATHS
+    ${CMAKE_CURRENT_LIST_DIR}/../../../@CMAKE_INSTALL_INCLUDEDIR@ # Install tree
```
Comment (allisonvacanti) [inline cub/cmake/cub-header-search.cmake.in:6]: Quotes

**Review by allisonvacanti — CHANGES_REQUESTED** [review]
Aside from some minor style points mentioned inline, this looks good to me.

**Comment by robertmaynard** [comment]
> Aside from some minor style points mentioned inline, this looks good to me.

All style issues are now resolved.

**Review by allisonvacanti — APPROVED** [review]

**Pull request merged**